### DISCUSSION TITLE
Kubernetes Executor wastefully creates pods for EmptyOperator tasks

### DISCUSSION THREAD

**Discussion by buu-nguyen** [apache/airflow#36477] [General]
### Description

Kubernetes Executor currently spins up a whole worker pod to execute an empty task. We know Airflow's EmptyOperator doesn't actually do work - it's there to help to organize/group tasks and very useful for that purpose.

### Use case/motivation

For complex DAGs, skipping the worker creation process for empty tasks could significantly improve execution time and resource use in the cluster. Could the executor work more like the local executor when dealing with the dummy tasks? (that way, they could be essentially instant and not interfere with the actual worker pool)

### Related issues

#7871

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by dirrao** [comment]
@buu-nguyen 
Can you share the airflow version and Kubernetes provider version where you are seeing this issue? I haven't seen this issue in the latest version.

**Comment by potiuk** [comment]
Yep . @dirrao is right. See the issue you are quoting - short-circuiting EmptyOperator has been implemented long time ago. Converted it into discussion until more details are provided and some evidences. showing what's going on.
